torch/distributions.py

```diff
@@ -41,7 +41,9 @@ class Distribution(object):
     r"""
     Distribution is the abstract base class for probability distributions.
     """
-
+    def __init__(self, reparameterized = False):
+        self.reparameterized = reparameterized
```

Comment:
I think this should be a per-class property. You can eliminate the `__init__` method and simply set
```py
class Distribution(object):
    reparameterized = False
    ...
class Normal(Distribution):
    reparameterized = True
    ...
```